docs/smart-contracts/entrypoints.md

```diff
@@ -12,77 +12,66 @@ The entrypoints in a Tezos smart contract must meet these specifications:
--->
+## Implementation details
 
+Even though your higher-level code may have separate codeblocks for each entrypoint, the compiled Michelson code uses a single codeblock with a single entrypoint, known as the default entrypoint.
+This default entrypoint uses the parameter that clients pass to decide which code to run.
 
-<!-- TODO old docs say "if this entrypoint exists" ; doesn't it always exist? -->
-<!--
-  - `default` doesn't take any parameter (or more specifically, its parameter is of type `Unit`). If this entrypoint exists, it will be executed any time the contract is called without specifying any entrypoint or parameter. This is in particular the case when a user or a contract simply sends some tez to the contract, as you do when sending tez to a user.
--->
+For example, when you compile the contract in the tutorial [Create a smart contract](../tutorials/smart-contract) to Michelson, its first line defines the parameter type that the contract accepts:
 
+```
+parameter (or (unit %reset) (or (int %decrement) (int %increment)))
+```
 
-## Entrypoint implementation
+To call the `reset` entrypoint, clients technically call the default entrypoint and pass the Michelson-encoded parameter `Left Unit`.
+This parameter value means that the left value of the parameter type, which is annotated `%reset`, is set to the value `Unit`, which means no value.
+The Michelson code uses the `IF_LEFT` command to check if the left value of the parameter is defined and if so, runs the `reset` entrypoint code.
 
-Internally, entrypoints are implemented using a variant data type.
-The contract contains a single piece of logic that branches based on the value of that variant.
-In Michelson, the different values of the variant are annotated with the name of the corresponding entrypoint.
-<!-- TODO link to variants -->
+Similarly, to call the `decrement` entrypoint and pass 5, clients call the default entrypoint and pass `Right (Left 5)`.
+This parameter value puts the value 5 in the integer value that is annotated `%decrement`.
+Then the code branches based on this parameter value.
 
-When calling a smart contract, senders provide either the full parameter as a variant or specify the name of the entrypoint and the corresponding parameter value.
+Developers need to know about the default entrypoint only when they encode parameters for smart contracts manually.
+Most Tezos clients, including Octez and Taquito, encode parameters automatically.
+Working from the previous example, they convert a call to the `increment` entrypoint with the parameter 5 as a call to the default entrypoint with the parameter `Right (Right 5)`.
 
-Other languages have different ways of indicating entrypoints.
+Different languages have different ways of indicating entrypoints.
 For information about coding entrypoints in specific languages, see these links:
 
 - Michelson: [Entrypoint](https://tezos.gitlab.io/active/michelson.html#entrypoints)
```

Comment:
maybe add a note on the octez-cli ? you can point to a specific entrypoint on arguments and just pass the expected payload without LEFT RIGHT stuff

Reply:
Yeah, octez may be a good example. How's this?

https://github.com/trilitech/tezos-developer-docs/pull/201/commits/daa013dc23add3692ba407e2ce8366ce2f34c695